specs/agents/metrics-health.md

```diff
@@ -0,0 +1,133 @@
+## Agent Health Metrics
+
+Agents SHOULD expose metrics which allow the analysis of the internal state of the agent, especially including the [reporting/transport mechanism](transport.md).
+
+Many of these metrics are `COUNTER`s: a `COUNTER` is a monotonically increasing metric.
+It is periodically reported by sending the difference of the counter-value now vs its value at the time of the previous report.
+In order to save elasticsearch disk space, zero-increases are not sent.
```

Comment:
I know I'm late to this game, sorry @JonasKunz ;-)
Is the main reason for using counters here (instead of gauges) to potentially save disk space?
What if those events are dropped though? Doesn't this skew the information?

Reply:
The main reason is that it is currently difficult to visualize cumulative counter metrics in kibana. E.g. for showing the `agent.events.total` metric in Kibana, the query needs to first compute the increase of the counter for each `event_type` and afterwards sum them up.

There has been an offline discussion about this topic with @axw , where we felt like going with delta-based counters is fine for now. We did a very small brainstorming on possibilities for improving the reliability:
 * Buffering failed events and retry them on agent side
 * Switching to cumulative counters when TSDB is ready (by creating new metrics and deprecating the previous ones)
Both have pros and cons, we might even do a mixture of both.

Reply:
I had a feeling that this was already discussed somewhere.
Thanks for the clarification!